Viewer/src/viewer/viewer.ts

```diff
@@ -653,6 +672,14 @@ export abstract class AbstractViewer {
 
         // create a new template manager for this viewer
         this.sceneManager = new SceneManager(this.engine, this._configurationContainer, this.observablesManager);
+        
+        this.sceneManager.onEnteringVRObservable.add(() => {
+            this.onEnteringVRObservable.notifyObservers(this);
+        });
+        
+        this.sceneManager.onExitingVRObservable.add(() => {
+            this.onExitingVRObservable.notifyObservers(this);

```

Comment:
Can you remind me all the places where there is an enteringVR event?

- vrHelper
- sceneManager
- observableManager
- viewer

Are them all right? Can you clarify why the sceneManager and observableManager ones are needed.

Reply:
After addressing this comment (https://github.com/BabylonJS/Babylon.js/pull/4909#discussion_r208828438), the onEnteringVR observable exists in the VRExperienceHelper and ObservablesManager only. 

When VR mode is entered through VRExperienceHelper.EnterVR(), SceneManager is notified and it then notifies the observers of ObservablesManager.onEnteringVRObservable. The Viewer is one of these observers and is therefore also notified when this happens. The same flow applies to exiting VR as well.

The ObservablesManager ones are needed because they act as the linked between the SceneManager and the Viewer.